src/main/java/core/basesyntax/PurchaseOperation.java

```diff
@@ -0,0 +1,15 @@
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    private final Map<String, Integer> fruitStorage;
+
+    public PurchaseOperation(Map<String, Integer> fruitStorage) {
+        this.fruitStorage = fruitStorage;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        fruitStorage.put(transaction.getFruit(), 
+                fruitStorage.getOrDefault(transaction.getFruit(), 0) - transaction.getQuantity());
+    }
```

Comment:
The `PurchaseOperation` class directly manipulates the `fruitStorage` map, which can lead to negative values if a purchase quantity exceeds the available quantity in storage. According to the checklist, you should check the result balance before saving it in the Storage and it should be positive. Throw a `RuntimeException` in case the balance is negative.